typings/index.d.ts

```diff
@@ -24,77 +26,72 @@ export enum StateChangeTypes {
   controlledPropUpdatedSelectedItem = '__autocomplete_controlled_prop_updated_selected_item__',
 }
 
-export interface DownshiftProps {
-  defaultSelectedItem?: any
+export interface DownshiftProps<Item> {
+  defaultSelectedItem?: Item
   defaultHighlightedIndex?: number | null
   defaultInputValue?: string
   defaultIsOpen?: boolean
-  itemToString?: (item: any) => string
-  selectedItemChanged?: (prevItem: any, item: any) => boolean
-  getA11yStatusMessage?: (options: A11yStatusMessageOptions) => any
+  itemToString?: (item: Item) => string
+  selectedItemChanged?: (prevItem: Item, item: Item) => boolean
+  getA11yStatusMessage?: (options: A11yStatusMessageOptions<Item>) => string
   onChange?: (
-    selectedItem: any,
-    stateAndHelpers: ControllerStateAndHelpers,
+    selectedItem: Item,
+    stateAndHelpers: ControllerStateAndHelpers<Item>,
   ) => void
   onSelect?: (
-    selectedItem: any,
-    stateAndHelpers: ControllerStateAndHelpers,
+    selectedItem: Item,
+    stateAndHelpers: ControllerStateAndHelpers<Item>,
   ) => void
   onStateChange?: (
-    options: StateChangeOptions,
-    stateAndHelpers: ControllerStateAndHelpers,
+    options: StateChangeOptions<Item>,
+    stateAndHelpers: ControllerStateAndHelpers<Item>,
   ) => void
   onInputValueChange?: (
     inputValue: string,
-    stateAndHelpers: ControllerStateAndHelpers,
+    stateAndHelpers: ControllerStateAndHelpers<Item>,
   ) => void
   stateReducer?: (
-    state: DownshiftState,
-    changes: StateChangeOptions,
-  ) => StateChangeOptions
+    state: DownshiftState<Item>,
+    changes: StateChangeOptions<Item>,
+  ) => StateChangeOptions<Item>
   itemCount?: number
   highlightedIndex?: number
   inputValue?: string
   isOpen?: boolean
-  selectedItem?: any
-  render?: ChildrenFunction
-  children?: ChildrenFunction
+  selectedItem?: Item
+  render?: ChildrenFunction<Item>
+  children?: ChildrenFunction<Item>
   id?: string
   environment?: Environment
   onOuterClick?: () => void
   onUserAction?: (
-    options: StateChangeOptions,
-    stateAndHelpers: ControllerStateAndHelpers,
+    options: StateChangeOptions<Item>,
+    stateAndHelpers: ControllerStateAndHelpers<Item>,
   ) => void
 }
 
 export interface Environment {
-  addEventListener: (type: string, cb: Function) => void
-  removeEventListener: (type: string, cb: Function) => void
+  addEventListener: typeof window.addEventListener
+  removeEventListener: typeof window.removeEventListener
   document: Document
 }
 
-export interface Document {
-  getElementById: (id: string) => HTMLElement
-}
-
-export interface A11yStatusMessageOptions {
+export interface A11yStatusMessageOptions<Item> {
   highlightedIndex: number | null
-  highlightedValue: any
```

Comment:
Removed this, because I haven't found any other mentions of this property

Reply:
As long as @kentcdodds is cool with this, we should remember to add this to the changelog.

Reply:
Yeah, I think that must have been a mistake. That isn't used anywhere in the codebase.